python/lsst/pipe/tasks/postprocess.py

```diff
@@ -789,6 +791,167 @@ def _makeArgumentParser(cls):
+        dimensions=("instrument", "visit", "detector"),
+        deferLoad=True,
+        multiple=True,
+    )
+    visitSummary = cT.Output(
+        doc="Consolidated visit-level exposure metadata",
+        name="visitSummary",
+        storageClass="ExposureCatalog",
+        dimensions=("instrument", "visit"),
+    )
+
+
+class ConsolidateVisitSummaryConfig(pipeBase.PipelineTaskConfig,
+                                    pipelineConnections=ConsolidateVisitSummaryConnections):
+    """Config for ConsolidateVisitSummaryTask"""
+    pass
+
+
+class ConsolidateVisitSummaryTask(pipeBase.PipelineTask, pipeBase.CmdLineTask):
+    """Task to consolidate per-detector visit metadata
+    """
+    _DefaultName = "consolidateVisitSummary"
+    ConfigClass = ConsolidateVisitSummaryConfig
+
+    @classmethod
+    def _makeArgumentParser(cls):
+        parser = ArgumentParser(name=cls._DefaultName)
+
+        parser.add_id_argument("--id", "calexp",
+                               help="data ID, e.g. --id visit=12345",
+                               ContainerClass=VisitDataIdContainer)
+        return parser
+
+    def writeMetadata(self, dataRef):
+        """No metadata to write.
+        """
+        pass
+
+    def writeConfig(self, butler, clobber=False, doBackup=True):
+        """No config to write.
+        """
+        pass
+
+    def runDataRef(self, dataRefList):
+        self.isGen3 = False
+
+        visit = dataRefList[0].dataId['visit']
+
+        self.log.debug("Concatenating metadata from %d per-detector calexps (visit %d)" %
+                       (len(dataRefList), visit))
+
+        expCatalog = self.run(visit, dataRefList)
+
+        dataRefList[0].put(expCatalog, 'visitSummary', visit=visit)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        self.isGen3 = True
+
+        dataRefs = butlerQC.get(inputRefs.calexp)
+        visit = dataRefs[0].dataId.byName()['visit']
+
+        self.log.debug("Concatenating metadata from %d per-detector calexps (visit %d)" %
+                       (len(dataRefs), visit))
+
+        expCatalog = self.run(visit, dataRefs)
+
+        butlerQC.put(expCatalog, outputRefs.visitSummary)
+
+    def run(self, visit, dataRefs):
+        """Make a combined exposure catalog from a list of dataRefs.
+
+        Parameters
+        ----------
+        visit : `int`
+            Visit identification number
+        dataRefs : `list`
+            List of calexp dataRefs in visit.  May be list of
+            `lsst.daf.persistence.ButlerDataRef` (Gen2) or
+            `lsst.daf.butler.DeferredDatasetHandle` (Gen3).
+
+        Returns
+        -------
+        visitSummary : `lsst.afw.table.ExposureCatalog`
+            Exposure catalog with per-detector summary information.
+        """
+        schema = afwTable.ExposureTable.makeMinimalSchema()
+        schema.addField('visit', type='I', doc='Visit number')
+        schema.addField('ccd', type='I', doc='Detector number')
```

Comment:
As this would require an additional loop over the table whenever you use it, I prefer to pull it out now when we already are incurring the loop overhead.